server/index.ts

```diff
@@ -94,6 +96,18 @@ async function start(id: number, disconnect: () => void) {
   onerror(app);
   app.on("error", requestErrorHandler);
 
+  // set a global rate limiter map which maps route path
```

Comment:
I think functionally this is all good but it has all ended up quite scattered around the codebase. It would be nice to keep everything rate-limiter related centralized in a single module.

There isn't a great existing pattern for this, for now perhaps just a file `server/RateLimiter.ts` that can contain the mapping, default limiter, and shutdown handler rather than being defined on context. Something like:

```
class RateLimiter {
  // mapping internal to class
  
  static getRateLimiter(path: string) {
    // returns default or custom limiter
  }
  
  static setRateLimiter(path, config) {
    // sets mapping
  }
  
  static shutdownHandler() {
   // logic from shutdown hook
  }
}
```

Reply:
Sure! Will iterate to organize it.

Reply:
Fixed! Thanks for raising this :raised_hands: